src/scripts/store/actions.js

```diff
@@ -1,41 +1,83 @@
+/* eslint-disable jsdoc/require-returns-type */
+
+/**
+ * @typedef {import('./index').Notice} Notice
+ */
+
+/**
+ * Action creator to hydrate a notice from the store.
+ *
+ * @param {Notice[]} payload The notices to hydrate.
+ * @return A redux action.
+ */
 export const hydrate = ( payload ) => {
 	return {
-		type: 'HYDRATE',
+		type: /** @type {'HYDRATE'} */ ( 'HYDRATE' ),
```

Comment:
This is weird but required for it to be type `'HYDRATE'` not `string`. Otherwise the type union in action types doesn't work right in the reducer.